.rubocop.yml

```diff
@@ -10,5 +10,8 @@ Metrics/PerceivedComplexity:
 Metrics/MethodLength:
   Max: 20
 
+Metrics/LineLength:
+  Max: 150
```

Comment:
raise UnexpectedAttributes, "Unexpected Attributes: #{remaining_attrs.join(', ')}" if remaining_attrs.size > 0

I think this is a perfect case were three lines is significantly more readable. A long line with a condition at the end is difficult to see the control flow. The following is much more readable IMO:

    if remaining_attrs.size > 0
      raise UnexpectedAttributes, "Unexpected Attributes: #{remaining_attrs.join(', ')}"
    end

Also if the UnexpectedAttributes constructor is customized, it need only take in the attributes as a list and the class itself can prepare the message permitting:

     raise UnexpectedAttributes(remaining_attrs) if remaining_attrs.size > 0

I would argue that's totally worth having some line limit for.